src/common/host_device_vector.h

```diff
@@ -250,11 +250,19 @@ class HostDeviceVector {
 
   /*!
    * \brief Specify memory distribution.
-   *
-   *   If GPUSet::Empty() is used, all data will be drawn back to CPU.
    */
-  void Reshard(const GPUDistribution& distribution) const;
-  void Reshard(GPUSet devices) const;
+  void Shard(const GPUDistribution &distribution) const;
+  void Shard(GPUSet devices) const;
+
+  /*!
+   * \brief Pull data back to CPU.
+   */
+  void Unshard() const;
+
+  /*!
+   * \brief Change memory distribution.
+   */
+  void Reshard(const GPUDistribution &distribution) const;
```

Comment:
A parameter to `Reshard()` indicating whether we want to preserve the data, or just to change the distribution of the object, may be useful. However, this is up to you, depending on the actual use case.

I would also avoid marking `Reshard()` `const`.

Reply:
Done.